torchaudio_contrib/layers.py

```diff
@@ -269,7 +285,7 @@ def __repr__(self):
 
 
 def Spectrogram(fft_len=2048, hop_len=None, frame_len=None,
-                window=None, pad=0, pad_mode="reflect", power=1., **kwargs):
+                window=None, pad=0, pad_mode="reflect", power=1., mono=True, **kwargs):
```

Comment:
* Couldn't it be better not to automate this? 
* Also, if we follow the discussion we had at https://github.com/keunwoochoi/torchaudio-contrib/issues/31#issuecomment-476786384, maybe something like `to_mono`, `downmix_to_mono`, `downmix` rather than `mono`? We mainly talked about `resample_to`, but the same philosophy can be applied here.

Reply:
Or, just be lazy and don't change `Spectrogram` in this commit, finish the downmix itself first?

Reply:
I agree that this shouldn't be the default. (The naming is another issue, but in #37 you suggested we can change anything later.)

Reply:
> I agree that this shouldn't be the default. 

this is risky since most spectrogram based tasks that do not output full spectrograms or masks but rather do classification etc would usually read stereo, downmix and perform the model on single channel. If the default downmix would be performed in the time domain using the mean, this is clearly wrong. Since in the `Spectrogram` layer we have control over the full pipeline (audio>stft>complexnorm>downmix), IMO it makes sense to put this default here instead of e.g. in the `ComplexNorm`.

Reply:
From a `Spectrogram` I would expect that it calculates the `Spectrogram`. If I put in a multi-channel signal, I would expect spectrograms for every channel. So, I would also vote for not making downmix the default behavior.

Reply:
okay, I mixed up "default should not be `mono=True`" with "default should not be doing it in the Spectrogram domain".  I will set it to `False` and implement the other suggested changes soon

Reply:
I'd even suggest removing `mono=` at all here. First of all, computing a spectrogram doesn't need to be merged with downmixing. Then there's still an issue how/where the downmix is done. Especially, if we're gonna have two downmixing methods (one on waveform, the other on spectrogram), I think we should encourage people to use them explicitly.

Reply:
Also, the bottom line is - let's discuss it in a separate PR.

Reply:
Lets use #54 for this